guide/token-setup-and-validation.md

```diff
@@ -0,0 +1,351 @@
+# Token setup and validation
+
+For an overview of the different token formats, see: [Link to token formats]
+
+This guide will be divided into 2 parts: token generation and token validation.
+
+In OpenID Connect there are three types of tokens: access tokens, id tokens, and refresh tokens. (link). When this guide refers to _tokens_ it is referring to access tokens.
+
+Authorization servers are responsible for token generation. Clients (server app or web app) request tokens and then use tokens to request resources. For example, a javascript web application may make api calls that require authorization, so the token is sent along in the header of every request.
+
+Token validation needs to be configured for servers that have api endpoints that require authorization. This could be authorization servers or standalone servers, called resource servers. An example authorization server api endpoint may be '/api/User' that returns the current user. A resource server api endpoint for a note-taking app may be '/notes' that returns the user's notes.
+
+Below shows code snippets for token generation and token validation.
+
+
+# Default configuration: Opaque tokens
+
+
+## Default token generation
+
+### Authorization server
+```c#
+// Startup.cs
+
+//ConfigureServices
+// The default token format uses ASP.NET Core Data Protection to encrypt and sign tokens.
+// If using ASP.NET Core Identity, this is called for you. It does not hurt if it's called again.
+    services.AddDataProtection();
```

Comment:
`services.AddDataProtection()` should be indirectly called by `services.AddOpenIddict().AddServer()`, `services.AddOpenIddict().AddValidation()` or anything using `services.AddAuthentication()`.

Did you encounter a case where it was necessary?